### PR TITLE
MM-22247: Space in content-type breaks integration with 3rd party apps

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

This PR fixes incoming webhooks that are using the url encoded content type with space
after the semicolon

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-22247

### CODE REVIEW

File: web/webhook.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func incomingWebhook(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	}()
 
-	if strings.Split(contentType, "; ")[0] == "application/x-www-form-urlencoded" {
+	if strings.Split(contentType, ";")[0] == "application/x-www-form-urlencoded" {
```
Comment (lieut-data) [inline web/webhook.go]: This seems to address the core issue, but I wonder if we might extend this solution to use https://golang.org/pkg/mime/#ParseMediaType and handle potentially other edge cases (i.e. case?)

Comment (ethervoid) [reply web/webhook.go]: Hum, good point. I'll add this, I checked for something like that but failed to find it :)

**Comment by ethervoid** [comment]
@lieut-data changes done :)

File: web/webhook.go [hunk 2]
```diff
@@ -28,7 +29,11 @@ func incomingWebhook(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	var err *model.AppError
 	incomingWebhookPayload := &model.IncomingWebhookRequest{}
-	contentType := r.Header.Get("Content-Type")
+	mediaType, _, mimerr := mime.ParseMediaType(r.Header.Get("Content-Type"))
```
Comment (lieut-data) [inline web/webhook.go]: nit: maybe `mimeErr`?

**Pull request merged**